Build failure: pcem

`pcem` fails to build on master:

```
       > In file included from slirp/debug.c:9:
       > slirp/debug.c: In function 'purgesocks':
       > slirp/slirp.h:70:25: error: implicit declaration of function 'close'; did you mean 'pclose'? [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration8;;]
       >    70 | # define closesocket(s) close(s)
       >       |                         ^~~~~
       > slirp/debug.c:388:9: note: in expansion of macro 'closesocket'
       >   388 |         closesocket(so->s);     //close the socket
       >       |         ^~~~~~~~~~~
       > make[1]: *** [Makefile:5825: slirp/pcem-debug.o] Error 1
       > make[1]: Leaving directory '/build/source/src'
       > make: *** [Makefile:367: all-recursive] Error 1
       For full logs, run 'nix log /nix/store/v6gnszsxlq1j16hsdlwbsrkpxgn0ha0m-pcem-17.drv'.
```

--- a/pkgs/by-name/pc/pcem/package.nix
+++ b/pkgs/by-name/pc/pcem/package.nix
@@ -43,6 +43,9 @@ stdenv.mkDerivation rec {
     ++ lib.optional withNetworking "--enable-networking"
     ++ lib.optional withALSA "--enable-alsa";
 
+  # Fix GCC 14 build
+  env.NIX_CFLAGS_COMPILE = "-Wno-error=implicit-function-declaration -Wno-error=incompatible-pointer-types";
+
   meta = with lib; {
     description = "Emulator for IBM PC computers and clones";
     mainProgram = "pcem";

